web/src/stream_create_subscribers_data.ts

```diff
@@ -50,6 +57,7 @@ export function add_user_ids(user_ids: number[]): void {
 export function remove_user_ids(user_ids: number[]): void {
     for (const user_id of user_ids) {
         user_id_set.delete(user_id);
+        undo_soft_remove_user_id(user_id);
```

Comment:
This might be a detail worth discussing in #design; my intuition is that it makes the most sense for a soft-removal state to be preserved when a pill gets removed for whatever reason, but maybe I'm missing something.

Reply:
https://chat.zulip.org/#narrow/stream/101-design/topic/Create.20channel.20-.20Remove.20button.2E/near/1842550